test/test_typed_array.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "sparrow/typed_array.hpp"
+
+#include "doctest/doctest.h"
+
+#include <iostream> // For Doctest
+#include <numeric>
+
+using namespace sparrow;
+
+using data_type_t = int32_t;
+using my_typed_array = typed_array<data_type_t>;
```

Comment:
Is there a way to use other types to test all the underlying layouts?

Reply:
I will templatize the tests 👍